Templating/TemplateGuesser.php

```diff
@@ -30,13 +30,22 @@ class TemplateGuesser
     protected $kernel;
 
     /**
+     * @var string[]
+     */
+    private $controllerPatterns;
+
+    /**
      * Constructor.
      *
-     * @param KernelInterface $kernel A KernelInterface instance
+     * @param KernelInterface $kernel             A KernelInterface instance
+     * @param string[]        $controllerPatterns Regexps extracting the controller name from its FQN.
      */
-    public function __construct(KernelInterface $kernel)
+    public function __construct(KernelInterface $kernel, array $controllerPatterns = array())
     {
+        $controllerPatterns[] = '/Controller\\\(.+)Controller$/';
```

Comment:
What about making this the default value of the config. It will allow to disable template guessing in the `Controller` directory is wanted (more flexible).